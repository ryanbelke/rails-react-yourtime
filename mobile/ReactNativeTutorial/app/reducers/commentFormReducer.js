import { fromJS } from 'immutable';

const UPDATE = 'COMMENT_FORM:UPDATE';
const RESET = 'COMMENT_FORM:RESET';

export const initialState = fromJS({
  meta: {},
});

const update = (state, action) => state.merge(action.payload);

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE:
      return update(state, action);
    case RESET: return initialState;
    default:
      return state;
  }
};

const updateCommentForm = (payload) => ({ type: UPDATE, payload });
const resetCommentForm = () => ({ type: RESET });

export const actions = {
  updateCommentForm,
  resetCommentForm,
};
