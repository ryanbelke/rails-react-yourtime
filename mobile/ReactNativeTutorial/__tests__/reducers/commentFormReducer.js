import commentFormReducer, { actions, initialState } from 'ReactNativeTutorial/app/reducers/commentFormReducer';

describe('commentFormReducer', () => {
  describe('updateCommentForm', () => {
    it('updates comment form with payload', () => {
      const payload = { author: 'Alexey', text: 'Random text' };
      expect(
        commentFormReducer(initialState, actions.updateCommentForm(payload))
      ).toMatchSnapshot();
    });
  });

  describe('resetCommentForm', () => {
    it('clears all values incl. meta', () => {
      const state = initialState.mergeDeep({
        meta: {
          errors: {
            author: 'Empty name',
          },
        },
        author: '',
        text: 'Random text',
      });
      expect(commentFormReducer(state, actions.resetCommentForm())).toMatchSnapshot();
    });
  });
});
