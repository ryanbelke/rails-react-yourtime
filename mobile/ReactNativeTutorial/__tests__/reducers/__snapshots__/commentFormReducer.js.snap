exports[`commentFormReducer resetCommentForm clears all values incl. meta 1`] = `
Object {
  "meta": Object {},
}
`;

exports[`commentFormReducer updateCommentForm updates comment form with payload 1`] = `
Object {
  "author": "Alexey",
  "meta": Object {},
  "text": "Random text",
}
`;
